src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -468,18 +468,15 @@ static grpc_error* cc_init_channel_elem(grpc_channel_element* elem,
           &error));
   grpc_channel_args_destroy(new_args);
   if (error != GRPC_ERROR_NONE) {
-    // Orphan the resolving LB policy and flush the exec_ctx to ensure
-    // that it finishes shutting down.  This ensures that if we are
-    // failing, we destroy the ClientChannelControlHelper (and thus
-    // unref the channel stack) before we return.
+    // Before we return, shut down the resolving LB policy, which destroys the
```

Comment:
How is ClientChannelControlHelper's destruction guaranteed by calling `chand->resolving_lb_policy.reset()`?

Reply:
`ResolvingLoadBalancingPolicy` does not create any refs to itself in its ctor, so resetting `chand->resolving_lb_policy` will remove the last ref and cause its destruction.  Since it owns the `ClientChannelControlHelper`, the helper will also be destroyed.  And the helper's destruction will release its ref on the channel stack.

This is essentially the same thing that was happening before this PR.  The only difference is that previously, we were (unnecessarily) bouncing into the combiner before releasing the ref to the LB policy, which is why we needed to flush the `ExecCtx` to make sure the destruction happened properly.  Now that we've removed that level of indirection, we don't need to flush the `ExecCtx`, but the same thing is still happening.

Reply:
That means we don't need `policy->channel_control_helper_.reset();` in `LoadBalancingPolicy::ShutdownAndUnrefLocked()` even before this PR, right?

Reply:
You might be right.  But it probably doesn't matter at this point. :)